Add new concealed mines

ErhardSteinhauer opened the pull request:
## About the PR
Replacement for upstream mine disguised as wet floor sign. New mines are less lethal (healthy unarmored target will not be put into crit by the explosion) and have both visual (blinking lights and pocking out wiring) and audio (ticking sounds) cues that can help player to spot a mine in advance. Concealed mine sprites are randomized.

## Why / Balance
More options for mappers, also makes mines less punishing.

## How to test
- Spawn newly added mines (look for the ones with Frontier suffix), step on them, get crippled, repeat.

## Media
![2025-3-04_18 23 21](https://github.com/user-attachments/assets/4500c1d9-5e1d-484e-aa58-35e9f8eb2ba7)

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
:cl: erhardsteinhauer
- add: Added new mines disguised as plushies, boxes and trash.

Clay500z commented:
Poor hampter

Houtblokje commented:
Scary

BluCottonCandy commented:
Please stop making expeditions so much harder and deadly with no positives...

the-hivequeen commented:
> Please stop making expeditions so much harder and deadly with no positives...

These are already in the game, they're making them easier to see.....

BluCottonCandy commented:
> > Please stop making expeditions so much harder and deadly with no positives...
> 
> These are already in the game, they're making them easier to see.....

No, currently just regular land mines exist.
We used to have wet floor sign mines (they went inert at some point) and now this is adding a new landmine variant that is only going to serve to make expeds even more annoying.

Given, that new guy you hired, will grab the landmine killing you all.

ErhardSteinhauer commented on Resources/Prototypes/_NF/Entities/Objects/Weapons/Bombs/mines.yml:
```diff
@@ -0,0 +1,735 @@
+    requiresAnchored: false
+    whitelist:
+      components:
+        - Actor
```
Right on

whatston3 commented:
Not suggesting to add it to this, but being able to disarm the mines and have them turn back into their original object (and maybe a separate mine if you're feeling particularly cheeky) would be neat: https://github.com/space-wizards/space-station-14/pull/33883

whatston3 commented on Resources/Prototypes/_NF/Entities/Objects/Weapons/Bombs/mines.yml:
```diff
@@ -0,0 +1,737 @@
+      params:
+        volume: 1
+    examinable: false
+  - type: TriggerOnProximity
```
```suggestion
  - type: TriggerOnProximity
    triggerSpeed: 0
```
With this it'll be an actual proximity bomb, and not need you to walk quickly over it.

whatston3 reviewed (commented):
This seems like a pretty reasonable implementation, unlikely to kill you unless you're pretty badly hurt.

The random sprites are nice, less predictable, and pretty reasonable.  The flash is a nice tell, and it's nice that the light is actually unshaded vs. the regular landmine (goddamn syndie drop pods, I swear)

ErhardSteinhauer commented:
> Not suggesting to add it to this, but being able to disarm the mines and have them turn back into their original object (and maybe a separate mine if you're feeling particularly cheeky) would be neat: [space-wizards/space-station-14#33883](https://github.com/space-wizards/space-station-14/pull/33883)

I gave it a try, and I'm not sure about this one. If the mines could be disarmed, then some players would be able to hoard them and cause quite a lot of damage later. And given how the random looks are implemented, I don't think that on disassembly I can return an item that matches the looks. And returning mine on disassembly again leads to hoarding.

whatston3 commented:
> I gave it a try, and I'm not sure about this one. If the mines could be disarmed, then some players would be able to hoard them and cause quite a lot of damage later. And given how the random looks are implemented, I don't think that on disassembly I can return an item that matches the looks. And returning mine on disassembly again leads to hoarding.

You can still pick them up to hoard them as-is.  Not _super practical_, because you need to hold onto the ticking thing and if you drop it it'll probably explode, but it _is_ an instant explosion grenade when thrown.
Another thing I found - they don't have a collision mask, so they won't collide with walls when thrown.  They'll pass right through.

ErhardSteinhauer commented:
> You can still pick them up to hoard them as-is. Not super practical, because you need to hold onto the ticking thing and if you drop it it'll probably explode, but it is an instant explosion grenade when thrown.

True, true. Changed so mines couldn't be picked up by players. Can be dragged, though. Also made so one could shoot (if aimed directly at the mine) the mine.

> Another thing I found - they don't have a collision mask, so they won't collide with walls when thrown. They'll pass right through.

Ack. Fixed in the last commit.

whatston3 reviewed (approved):
Looks good to me.
